ci/integTestGen/src/integTestGen.jl

```diff
@@ -0,0 +1,265 @@
+
+# Returns
+- `::AbstractVector{String}`: all packages, which have the search dependency
+
+"""
+function get_depending_projects(package_name::String, package_prefix::Union{AbstractString,Regex}, project_tree=PkgDependency.builddict(Pkg.project().uuid, Pkg.project()))::AbstractVector{String}
+    packages::AbstractVector{String} = []
+    visited_packages::AbstractVector{String} = []
+    traverse_tree!(package_name, package_prefix, project_tree, packages, visited_packages)
+    return packages
+end
+
+"""
+    traverse_tree!(package_name::String, package_prefix::Union{AbstractString,Regex}, project_tree, packages::AbstractVector{String}, visited_packages::AbstractVector{String})
+
+Traverse a project tree and add package to `packages`, which has the package `package_name` as dependency. Ignore all packages, which does not start with `package_prefix`.
+See [`get_depending_projects`](@ref)
+
+"""
+function traverse_tree!(package_name::String, package_prefix::Union{AbstractString,Regex}, project_tree, packages::AbstractVector{String}, visited_packages::AbstractVector{String})
+    for project_name_version in keys(project_tree)
+        # remove project version from string -> usual shape: `packageName.jl version`
+        project_name = split(project_name_version)[1]
+        # fullfil the requirements
+        # - package starts with the prefix
+        # - the dependency is not nothing (I think this representate, that the package was already set as dependency of a another package and therefore do not repead the dependencies)
+        # - has dependency
+        # - was not already checked
+        if startswith(project_name, package_prefix) && project_tree[project_name_version] !== nothing && !isempty(project_tree[project_name_version]) && !(project_name in visited_packages)
+            # only investigate each package one time
+            # assumption: package name with it's dependency is unique
+            push!(visited_packages, project_name)
+            for dependency_name_version in keys(project_tree[project_name_version])
+                # dependency matches, add to packages
+                if startswith(dependency_name_version, package_name)
+                    push!(packages, project_name)
+                    break
+                end
+            end
+            # independent of a match, under investigate all dependencies too, because they can also have the package as dependency
+            traverse_tree!(package_name, package_prefix, project_tree[project_name_version], packages, visited_packages)
+        end
+    end
+end
+
+"""
+    generate_job_yaml!(package_name::String, job_yaml::Dict)
+
+Generate GitLab CI job yaml for integration test of a give package.
+
+# Args
+- `package_name::String`: Name of the package to test.
+- `job_yaml::Dict`: Add generated job to this dict.
+"""
+function generate_job_yaml!(package_name::String, job_yaml::Dict)
```

Comment:
Depending on mutable global variables is unsafe and highly discouraged. Maybe pass `package_infos` as an argument.